orix/io/plugins/ctf.py

```diff
@@ -0,0 +1,357 @@
+    indexed:
+
+    * Oxford AZtec HKL: Phase == 0
+
+    Parameters
+    ----------
+    filename
+        Path and file name.
+
+    Returns
+    -------
+    xmap
+        Crystal map.
+    """
+    # Get file header
+    with open(filename, "r") as f:
+        [header, data_starting_row] = _get_header(f)
+
+    # Get phase names and crystal symmetries from header (potentially empty)
+    phase_ids, phase_names, symmetries, lattice_constants = _get_phases_from_header(
+        header
+    )
+    structures = []
+    for name, abcABG in zip(phase_names, lattice_constants):
+        structures.append(Structure(title=name, lattice=Lattice(*abcABG)))
+
+    # Read all file data
+    file_data = np.loadtxt(filename, skiprows=data_starting_row)
+
+    # Get vendor and column names
+    n_rows, n_cols = file_data.shape
+    vendor, column_names = _get_vendor_columns(header, n_cols)
+
+    # Data needed to create a CrystalMap object
+    data_dict = {
+        "euler1": None,
+        "euler2": None,
+        "euler3": None,
+        "x": None,
+        "y": None,
+        "phase_id": None,
+        "prop": {},
+    }
+    for column, name in enumerate(column_names):
+        if name in data_dict.keys():
+            data_dict[name] = file_data[:, column]
+        else:
+            data_dict["prop"][name] = file_data[:, column]
+
+    # Add phase list to dictionary
+    data_dict["phase_list"] = PhaseList(
+        names=phase_names,
+        space_groups=symmetries,
+        structures=structures,
+        ids=phase_ids,
+    )
+
+    # Set which data points are not indexed
+    if vendor in ["orix", "hkl"]:
+        not_indexed = data_dict["phase_id"] == 0
+        data_dict["phase_id"][not_indexed] = -1
+
+    # Set scan unit
+    scan_unit = "um"
+    data_dict["scan_unit"] = scan_unit
+
+    # Create rotations
+    data_dict["rotations"] = Rotation.from_euler(
+        np.column_stack(
+            (data_dict.pop("euler1"), data_dict.pop("euler2"), data_dict.pop("euler3"))
+        ),
+        degrees=True,
+    )
+
+    return CrystalMap(**data_dict)
+
+
+def _get_header(file: TextIOWrapper) -> List[str]:
+    """Return the first lines above the mapping data and the data starting row number
+    in an .ctf file.
+
+    Parameters
+    ----------
+    file
+        File object.
+
+    Returns
+    -------
+    header
+        List with header lines as individual elements.
+    data_starting_row
+        The starting row number for the data lines
+    """
+    all_data = [line.rstrip() for line in file.readlines()]
+
+    phase_num_row = 0
+    for line in all_data:
+        if "Phases" in line:
+            phases_num_line = line
+            break
+        phase_num_row += 1
+
+    phase_num = int(phases_num_line.split("\t")[1])
+    header = all_data[: (phase_num_row + phase_num + 1)]
+    data_starting_row = phase_num_row + phase_num + 2
+    return header, data_starting_row
```

Comment:
To avoid reading all lines in the file, I believe we can read all lines until we get a match for "Phase tab X tab Y":

```suggestion
    header = []
    line = file.readline()
    i = 0
    while not line.startswith("Phase\tX\tY"):
        header.append(line.rstrip())
        i += 1
        line = file.readline()
    return header, i + 1
```